# Rank Prediction 

It was a warm summer evening in Ahmedabad in 2015, and my brother and I were sitting in our room, lost in thought. I was thinking about my future, contemplating the results of my Board exams and JEE Main exams, which had just been released.

As I sat there, deep in thought, my brother broke the silence.

"What do you think about which college you're going to apply to?" he asked, as he continued to work on his coding assignments in Java.

"I don't know," I replied. "I'll apply to whichever ones I'm eligible for."

My brother seemed to sense my uncertainty and pressed further.

"What do you think your merit number will be based on your Board exam and JEE Main scores?" he asked.

"I don't know, brother," I replied again. "But I can calculate it."

My brother was intrigued by my response and asked me to develop an algorithm to calculate my merit number. I thought for a moment and then i tought i can calculate using a linear equation with one variable.

"We have last year's data," I explained, referring to the student 2014.pdf file we had. "It shows which merit number each student received based on their Board exam and JEE Main scores. So, I  calculated the slope 'm' and 'c' on paper from this data and then determine my rank accordingly."

My brother was impressed by my idea and decided to write a Java program to predict my rank using the linear equation with one variable. We worked together, and with his coding skills and my suggested algorithm, we were able to determine my predicted rank.

## Data Information

We possess the data of students from the year 2014, which is documented in a file named 'student 2014.pdf'.

Every year around 56,000 thousand student enroll in Gujarat Located engineering colleges in Gujarat,India.

This data indicates the percentile ranking achieved by each student based on their scores in the JEE Main and Board examinations.

The Merit Percentile is a composite score derived from 60% of the Board percentile and 40% of the JEE Main percentile.

The ranking of the students is determined based on the Merit Mark, which is used to assign their respective ranks. Subsequently, the students select engineering institutes in Gujarat, India based on their rank.

## Take values 

I'm taking two records from "student 2014.pdf".



| Record | Merit Number | Merit Percentile | Board Percentile | JEE Main percentile |
|----------|----------|----------|----------|----------|
|  1 |10000 | 84.84442058 | 87.95556955 | 80.17769713 |
|  2 | 20000 | 71.50631167 | 70.8503965 | 72.49018442 |



## Calculate slope "m"

"I will be using the linear equation, '**y = mx + c**', to predict the rank of the candidates. 

The assumption made is that there is a linear relationship between merit percentile and merit number, ( i think it comes from my previous experience studying algebra for two years in high school before enrolling in engineering college). 

I took two records, one with 'Merit Number 10000' and the other with 'Merit Number 20000'.

Assigning 

y1 = 10000, x1 = 84.84442058


and 

y2 = 20000, x2 = 71.50631167


now want to calculate "m" which slope, 

m = (y2-y1) / (x2-x1)


In [55]:
# asigning variable for first record.
y1 = 10000
x1 = 84.84442058

# asigning variable for second record.
y2 = 20000
x2 = 71.50631167

In [56]:
# Calculating slope
m = (y2-y1)/(x2-x1)
print("Slope m : "  + str(m))

Slope m : -749.7314699914231


## Calculate y-interscept

Got slope value "m" from last year student records data.

now calculate y-intercept

**y = m*x + c** 

we can write above equation as below

**c = y - m*x**

In [57]:
# with help first we calculate "c" y-intercept
c = y1 - m*x1
print(c)

73610.53216201396


As we got slope "m" and y-intercept "y" values from student 2014 data.

now we can predict my rank as we got all the values and data.

As in year 2015, my score is as below

Board Percentile : 67.22

Jee Main Percentile : 93.42

Merit Percenile = 60% Board Percentile + 40% Jee Main Percentile

Merit percentile = 77.70

## Predicting My Rank

In [58]:
# x is my merit percentile 

# y is my merit number 

# decreased the value of "C" by 25000 because in year 2014 59000 thousand student took admission in 
# engineering college based in Gujarat, India and where as in year 2015 around 56000 thousand 
# students going to take admission. ( i just did because i think 3000 student change slope little bit
# but the difference in y-intercept could be anywhere between 20000 to 30000. It's just a random assumption )

c = -2500 + 73610.53216201396


x = 77.70

y = round(m*x + c)

print("My Predicted Merit Number is : "+ str(y))

My Predicted Merit Number is : 12856



I searched manually in "college 2014.pdf" which college i can get.

My criteria of choosing this colleges was as follows:

1. My predicted rank is 12856
2. My father advised me to go to Government Engineering College, because they have good faculties, infrastructure and very very low tution fees.
3. I was intrested in Electronics that time.
4. I wanted to stay in same city i.e Ahmedabad.


So, based on my predicted rank and criteria mentioned above, i shortlisted two college and courses is as follow:

1. LDCE, Electronics and Communication Engineering
2. VGEC, Electronics and Communication Engineering


Two weeks after this predicted rank i got **my actual rank given by ACPC body that was 12753**.



In [59]:
y_a = 12753

diff = abs(y - y_a)
print("The difference between predicted rank and actual rank was : " + str(diff))

The difference between predicted rank and actual rank was : 103


As i got my actual rank, i applied in these two colleges and managed to get admission in 

**VGEC, Electronics and Communication Engineering on August 2015.**

In Sept 2020, I was successfully graduated from this Government Engineering College.

## Thank You.